### PR TITLE
fix: avif encoding is too slow

### PR DESCRIPTION
**Pull request by yisibl**
Let's be consistent with cavif's configuration, speed=1 or 2 are unbelievably slow.

https://github.com/kornelski/cavif-rs#usage

> - `--quality=n` — Quality from 1 (worst) to 100 (best), the default value is 80. The numbers have different meaning than JPEG's quality scale. Beware when comparing codecs. There is no lossless compression support.
>
> - `--speed=n` — Encoding speed between 1 (best, but slowest) and 10 (fastest, but a blurry mess), the default value is 4. Speeds 1 and 2 are unbelievably slow, but make files ~3-5% smaller. Speeds 7 and above degrade compression significantly, and are not recommended.

It's much faster now:
```shell
$ node example/anime-girl.js
✨ png done in 869.3279090076685 ms
✨ avif done in 2122.1105709969997 ms
```

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -45,7 +45,7 @@ const MIME_PNG: &str = "image/png";
 const MIME_JPEG: &str = "image/jpeg";
 const MIME_AVIF: &str = "image/avif";
 
-const DEFAULT_IMAGE_QUALITY: u8 = 92;
+const DEFAULT_IMAGE_QUALITY: u8 = 80;
```
Comment (Brooooooklyn) [inline src/lib.rs]: 这里好像会影响到 `toDataURL`, 按照规范 `toDataURL` 的 quality 默认值是 92: https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL

Comment (yisibl) [reply src/lib.rs]: 好的我看到了，在 Blink 中：
- jpeg 默认 92： https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/platform/image-encoders/image_encoder.cc;l=85;drc=81c6f843fdfd8ef660d733289a7a32abe68e247a
- webp 默认 80：https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/platform/image-encoders/image_encoder.cc;l=100;drc=81c6f843fdfd8ef660d733289a7a32abe68e247a

File: src/lib.rs [hunk 2]
```diff
@@ -324,7 +334,7 @@ fn to_data_url(ctx: CallContext) -> Result<JsString> {
   let mime = mime_js.as_str()?;
   let quality = if ctx.length < 2 {
     // https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL
-    DEFAULT_IMAGE_QUALITY
+    DEFAULT_JPEG_QUALITY
```
Comment (yisibl) [inline src/lib.rs:345]: 这里需要区分  WebP 和 JPEG 吗

File: src/lib.rs [hunk 3]
```diff
@@ -347,7 +357,7 @@ fn to_data_url_async(ctx: CallContext) -> Result<JsObject> {
...
```
Comment (yisibl) [inline src/lib.rs:370]: 这里需要区分  WebP 和 JPEG 吗

Comment (Brooooooklyn) [reply src/lib.rs:370]: 感觉需要区分一下，按照浏览器实现来吧

Comment (yisibl) [reply src/lib.rs:370]: 这里这样区分吗？
```rust
if mine == "image/webp" {
    DEFAULT_WEBP_QUALITY
 }
```

File: index.js [hunk 4]
```diff
@@ -221,16 +221,28 @@ function createCanvas(width, height, flag) {
...
 
+  // See also: https://github.com/kornelski/cavif-rs#usage
+  const avifDefaultQuality = 80
+  const avifConfig = {
+    // Quality from 1 (worst) to 100 (best), the default value is 80. The numbers have different meaning than JPEG's quality scale. Beware when comparing codecs. There is no lossless compression support.
+    quality: avifDefaultQuality,
+
+    // Calculate alphaQuality, this is consistent with cavif.
+    // https://github.com/kornelski/cavif-rs/blob/37847b95bb81d4cf90e36b7fab2c7fbbcf95abe2/src/main.rs#L97
+    alphaQuality: Math.min((avifDefaultQuality + 100) / 2, avifDefaultQuality + avifDefaultQuality / 4 + 2),
```
Comment (yisibl) [inline index.js:232]: 删除了 alphaQuality 选项，和 cavif 的计算保持一致。

File: __test__/draw.spec.ts [hunk 5]
```diff
@@ -928,6 +928,18 @@ test('toDataURL with quality', async (t) => {
   t.deepEqual(pngBuffer, await canvas.encode('jpeg', 20))
 })
 
+test.only('WebP toDataURL with quality', async (t) => {
```
Comment (Brooooooklyn) [inline __test__/draw.spec.ts]: remove

File: src/lib.rs (hunk 3 above)
Comment (Brooooooklyn) [reply src/lib.rs:370]: `mime == MIME_WEBP`

File: __test__/draw.spec.ts (hunk 5 above)
Comment (Brooooooklyn) [reply __test__/draw.spec.ts]: 嗯

Comment (yisibl) [reply __test__/draw.spec.ts]: done

File: src/lib.rs [hunk 6]
```diff
@@ -344,12 +360,14 @@ fn to_data_url(ctx: CallContext) -> Result<JsString> {
 #[js_function(2)]
 fn to_data_url_async(ctx: CallContext) -> Result<JsObject> {
   let mime_js = ctx.get::<JsString>(0)?.into_utf8()?;
-  let mime = mime_js.as_str()?;
-  let quality = if ctx.length < 2 {
-    // https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL
-    DEFAULT_IMAGE_QUALITY
-  } else {
+  let mut mime = mime_js.as_str()?;
```
Comment (Brooooooklyn) [inline src/lib.rs]: 不需要 mut

File: src/lib.rs [hunk 7]
```diff
@@ -321,12 +335,14 @@ fn data(ctx: CallContext) -> Result<JsBuffer> {
 #[js_function(2)]
 fn to_data_url(ctx: CallContext) -> Result<JsString> {
   let mime_js = ctx.get::<JsString>(0)?.into_utf8()?;
-  let mime = mime_js.as_str()?;
-  let quality = if ctx.length < 2 {
-    // https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL
-    DEFAULT_IMAGE_QUALITY
-  } else {
+  let mut mime = mime_js.as_str()?;
```
Comment (Brooooooklyn) [inline src/lib.rs]: 不需要 mut

File: src/lib.rs [hunk 8]
```diff
@@ -269,11 +280,14 @@ fn encode_sync(ctx: CallContext) -> Result<JsBuffer> {
 #[js_function(2)]
 fn to_buffer(ctx: CallContext) -> Result<JsBuffer> {
   let mime_js = ctx.get::<JsString>(0)?.into_utf8()?;
-  let mime = mime_js.as_str()?;
-  let quality = if ctx.length < 2 {
-    DEFAULT_IMAGE_QUALITY
-  } else {
+  let mut mime = mime_js.as_str()?;
```
Comment (Brooooooklyn) [inline src/lib.rs]: 不需要 mut

File: src/lib.rs (hunk 6 above)
Comment (yisibl) [reply src/lib.rs]: done

**Pull request merged**